pkg/gluetidb/glue.go

```diff
@@ -79,19 +105,63 @@ func (gs *tidbSession) CreateDatabase(ctx context.Context, schema *model.DBInfo)
 // CreateTable implements glue.Session.
 func (gs *tidbSession) CreateTable(ctx context.Context, dbName model.CIStr, table *model.TableInfo) error {
 	d := domain.GetDomain(gs.se).DDL()
-
+	query, err := gs.showCreateTable(table, dbName)
+	if err != nil {
+		return err
+	}
+	gs.se.SetValue(sessionctx.QueryString, query)
 	// Clone() does not clone partitions yet :(
 	table = table.Clone()
 	if table.Partition != nil {
 		newPartition := *table.Partition
 		newPartition.Definitions = append([]model.PartitionDefinition{}, table.Partition.Definitions...)
 		table.Partition = &newPartition
 	}
-
 	return d.CreateTableWithInfo(gs.se, dbName, table, ddl.OnExistIgnore, true)
 }
 
 // Close implements glue.Session.
 func (gs *tidbSession) Close() {
 	gs.se.Close()
 }
+
+func (gs *tidbSession) allocatorFor(tbl *model.TableInfo, dbName model.CIStr) (autoid.Allocator, error) {
+	db, ok := domain.GetDomain(gs.se).InfoSchema().SchemaByName(dbName)
+	if !ok {
+		return nil, errors.Errorf("failed to get schema of table %s", tbl.Name.String())
+	}
+	if alloc, ok := gs.allocs[db.ID]; ok {
+		return alloc, nil
+	}
+	alloc := gs.allocFactory(db.ID, autoid.AutoIncrementType)
+	gs.allocs[db.ID] = alloc
+	return alloc, nil
+}
+
+// showCreateTable shows the result of SHOW CREATE TABLE from a TableInfo.
+func (gs *tidbSession) showCreateTable(tbl *model.TableInfo, dbName model.CIStr) (string, error) {
+	table := tbl.Clone()
+	table.AutoIncID = 0
+	result := bytes.NewBuffer(make([]byte, 0, defaultCapOfCreateTable))
+	alloc, err := gs.allocatorFor(tbl, dbName)
+	if err != nil {
+		return "", err
+	}
+	// this can never fail.
+	_, _ = result.WriteString(brComment)
+	if err := executor.ConstructResultOfShowCreateTable(gs.se, tbl, alloc, result); err != nil {
```

Comment:
let's just pass `nil` here. the allocators are only used for reading out the auto-inc and auto-random IDs which we will explicitly reset anyway.

```suggestion
	if err := executor.ConstructResultOfShowCreateTable(gs.se, tbl, nil, result); err != nil {
```

Reply:
I do want to do that, but when this is `nil`, we will meet a panic if the table has auto increment columns...

Reply:
(at executor/show.go:907)

Reply:
@YuJencen what about `autoid.Allocators{}`

Reply:
`autoid.Allocators` doesn't implement `Allocator`. 😢

Reply:
Oops it was just [changed to `Allocators`](https://github.com/pingcap/tidb/blob/a99fdc098cb3bccc63ca74dad3d682b6f80a0fbd/executor/show.go#L722) in the latest master since pingcap/tidb#18217 👀

Reply:
OK, seems after we update tidb to newest master, we can give it a `autoid.Allcators{}`. 😄